VideoLocker/src/main/java/org/edx/mobile/http/HttpResponseStatusException.java

```diff
@@ -2,15 +2,14 @@
 
 import android.support.annotation.NonNull;
 
-public class HttpResponseStatusException extends RetroHttpException {
-    private final int statusCode;
+import retrofit.RetrofitError;
 
-    public HttpResponseStatusException(@NonNull Throwable cause, int statusCode) {
+public class HttpResponseStatusException extends RetroHttpException {
+    public HttpResponseStatusException(@NonNull RetrofitError cause) {
         super(cause);
-        this.statusCode = statusCode;
     }
 
     public int getStatusCode() {
-        return statusCode;
+        return getCause().getResponse().getStatus();
```

Comment:
`HttpResponseStatusException` can only be validly generated if there is a response. However, I suppose there should be an assertion of that in the constructor, and in the other specific subclasses on `RetroHttpException` as well.